`make test` produces inconsistent results under Python 3

I get different results each time I run `make test`, under <del>both Python 2 and 3</del> <ins>Python 3</ins>. For example:

```
(testimony3) [ichimonji10@beech:testimony]$ make test
178c178
< test_positive_login_1
---
> test_positive_login_2
180a181
> * Missing docstring.
182,186d182
< * Unexpected tokens:
<   Bug: 123456
<   Feture: Login - Positive
<   Statues: Manual
<   Types: Functional
188c184
< test_positive_login_2
---
> test_negative_login_5
191d186
< * Missing docstring.
194c189
< test_negative_login_5
---
> test_positive_login_1
197a193,197
> * Unexpected tokens:
>   Bug: 123456
>   Feture: Login - Positive
>   Statues: Manual
>   Types: Functional
make: *** [Makefile:13: test] Error 1
(testimony3) [ichimonji10@beech:testimony]$ make test
178c178
< test_positive_login_1
---
> test_negative_login_5
182,186d181
< * Unexpected tokens:
<   Bug: 123456
<   Feture: Login - Positive
<   Statues: Manual
<   Types: Functional
194c189
< test_negative_login_5
---
> test_positive_login_1
197a193,197
> * Unexpected tokens:
>   Bug: 123456
>   Feture: Login - Positive
>   Statues: Manual
>   Types: Functional
make: *** [Makefile:13: test] Error 1
(testimony3) [ichimonji10@beech:testimony]$ make test
177a178,183
> test_positive_login_2
> ---------------------
> 
> * Missing docstring.
> * Docstring should have at least assert, feature, test token(s)
> 
187,192d192
< 
< test_positive_login_2
< ---------------------
< 
< * Missing docstring.
< * Docstring should have at least assert, feature, test token(s)
make: *** [Makefile:13: test] Error 1
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,5 +1,9 @@
 sudo: false
 language: python
+python:
+    - "2.7"
+    - "3.4"
+    - "3.5"
 install: pip install -r requirements-dev.txt .
 script:
     - make lint

--- a/testimony/__init__.py
+++ b/testimony/__init__.py
@@ -277,7 +277,8 @@ def validate_docstring_report(testcases):
                 ))
                 invalid_tags_docstring_count += 1
             if issues:
-                result.setdefault(path, {})[testcase.name] = issues
+                result.setdefault(
+                    path, collections.OrderedDict())[testcase.name] = issues
                 invalid_docstring_count += 1
 
     if SETTINGS['json']:

